azkaban-common/src/main/java/azkaban/executor/container/watch/FlowStatusUpdatingListener.java

```diff
@@ -0,0 +1,317 @@
+package azkaban.executor.container.watch;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+import azkaban.Constants.ContainerizedDispatchManagerProperties;
+import azkaban.executor.AlerterHolder;
+import azkaban.executor.ExecutableFlow;
+import azkaban.executor.ExecutionControllerUtils;
+import azkaban.executor.ExecutorLoader;
+import azkaban.executor.ExecutorManagerException;
+import azkaban.executor.Status;
+import azkaban.executor.container.ContainerizedImpl;
+import azkaban.utils.Props;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheStats;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+import javax.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provides callback implementations of {@link AzPodStatusListener} for
+ * (1) Updating WebServer and database state of flows based on Pod status
+ * (2) Driving pod lifecycle actions, such as deleting flow pods in a final state.
+ */
+public class FlowStatusUpdatingListener implements AzPodStatusListener {
```

Comment:
How about FlowStatusManagerListener? Update is one action. Better to keep it generic.